exciting_environments/mass_spring_damper/mass_spring_damper_env.py

```diff
@@ -15,121 +18,179 @@ class MassSpringDamper(core_env.CoreEnvironment):
+    def _ode_solver_step(self, states, action, static_params):
+        """Computes states by simulating one step.
+
+        Args:
+            states: The states from which to calculate states for the next step.
+            action: The action to apply to the environment.
+            static_params: Parameter of the environment, that do not change over time.
 
-        env_states_norm = states_norm
-        force = force_norm*action_normalizer
-        env_states = env_state_normalizer * env_states_norm
-        args = (force, static_params)
+        Returns:
+            states: The computed states after the one step simulation.
+        """
+
+        env_states = states.physical_states
+        args = (action, static_params)
 
         def vector_field(t, y, args):
             deflection, velocity = y
-            force, params = args
-            d_velocity = (force[0] - params["d"]
-                          * velocity - params["k"]*deflection)/params["m"]
+            action, params = args
+            d_velocity = (action[0] - params.d
+                          * velocity - params.k*deflection)/params.m
             d_deflection = velocity
-            d_y = d_deflection, d_velocity[0]
+            d_y = d_deflection, d_velocity  # [0]
             return d_y
 
         term = diffrax.ODETerm(vector_field)
         t0 = 0
         t1 = self.tau
-        y0 = tuple(env_states)
+        y0 = tuple([env_states.deflection, env_states.velocity])
         env_state = self._solver.init(term, t0, t1, y0, args)
         y, _, _, env_state, _ = self._solver.step(
             term, t0, t1, y0, args, env_state, made_jump=False)
 
         deflection_k1 = y[0]
         velocity_k1 = y[1]
 
-        env_states_k1 = jnp.hstack((
-            deflection_k1,
-            velocity_k1,
-        ))
-        env_states_k1_norm = env_states_k1/env_state_normalizer
+        phys = self.PhysicalStates(
+            deflection=deflection_k1, velocity=velocity_k1)
+        opt = None  # Optional(something=...)
+        return self.States(physical_states=phys, PRNGKey=None, optional=None)
 
-        return env_states_k1_norm
+    @partial(jax.jit, static_argnums=0)
+    def init_states(self):
+        """Returns default initial states for all batches."""
+        phys = self.PhysicalStates(deflection=jnp.zeros(
+            self.batch_size), velocity=jnp.zeros(self.batch_size))
+        opt = None  # self.Optional(something=jnp.zeros(self.batch_size))
+        return self.States(physical_states=phys, PRNGKey=None, optional=opt)
 
     @partial(jax.jit, static_argnums=0)
-    def default_reward_func(self, obs, action):
-        return ((obs[0])**2 + 0.1*(obs[1])**2 + 0.1*(action[0])**2)
+    def default_reward_func(self, obs, action, action_constraints):
+        """Returns reward for one batch."""
+        reward = (obs[0])**2 + 0.1*(obs[1])**2 + 0.1 * \
+            (action[0]/action_constraints.force)**2
```

Comment:
```suggestion
        reward = ((obs[0])**2 + 0.1*(obs[1])**2 
                  + 0.1 * (action[0]/action_constraints.force)**2)
```

Brackets for line continuation. I like to split at the additions and IMO the operator should be the first element in the new line.